packages/edit-site/src/store/selectors.js

```diff
@@ -341,10 +341,16 @@ export const getCurrentTemplateTemplateParts = createRegistrySelector(
 			'wp_template_part',
 			{ per_page: -1 }
 		);
-		const templatePartsById = keyBy(
-			templateParts,
-			( templatePart ) => templatePart.id
-		);
+		const templatePartsById = templateParts
+			? // Key template parts by their ID.
+			  templateParts.reduce(
+					( newTemplateParts, part ) => ( {
+						...newTemplateParts,
+						[ part.id ]: part,
+					} ),
+					{}
+			  )
+			: {};
```

Comment:
How about a loop instead?

```suggestion
		const templatePartsById = {};
		if ( templateParts ) {
			for( const part of templateParts ) {
				templatePartsById[ part.id ] = part;
			}
		}
```

It's shorter and easier to scan.

Reply:
Same for other updated files.

Reply:
As explained in my above comment, I prefer the `reduce()` version because it fits better within the FP-style that's most common in the repo. I honestly see this as a very simple `reduce()` implementation, and a pretty obvious one, too. Also, I'm not particularly happy with mutating objects - it could introduce subtle bugs in the future, I've seen that before. And just in case that helps, I've added a comment to make the intention clear. So unless you have really strong feelings about this, I'd prefer keeping the `reduce()` version.

Reply:
I have semi-strong feeling about reduce vs loops:

* The loop is shorter
* I can immediately tell what it does without any explanatory comment.
* The reduce implementation required that explanatory comment which tells me it's not clear from just looking.

Personally, I always get confused by all but the simplest `reduce()` implementations and end up reading the code for a few minutes. There's just something unnatural about the reduce mental model for me specifically – I love FP but at the same time I tend to think imperatively :-)

Let's agree to disagree, I guess. I don't want to block your work so let's get a second opinion here. Maybe @jsnajdr or @gziolo or @mcsf  would like to chip in?

Reply:
IMO this is definitely a `reduce` use case and I agree with @tyxla that is a simple one too. The only change I'd recommend that might made it more readable could be to change the `previous value` to `accumulator` as we do in many places in our code base and in some places like change the name of the `current value` to be even more descriptive. For example this: `newTemplateParts, part` -> `accumulator, templatePart`

Reply:
Also wanted to note that `keyBy` can also be confusing to some and this includes me. I had to go to lodash docs to see what exactly is for.

Reply:
> Maybe @jsnajdr or @gziolo or @mcsf would like to chip in?

Seems a bit excessive to ping for this. :) As long as care is put into the flow and the naming of things, the question of looping vs. reducing doesn't matter that much. (But I admit that I [haven't always felt this way](https://lamda.blog/2017/01/15/of-vocabulary-and-contracts/).)

Just to troll a bit, well, why not an even higher-level construct?

```js
const templatePartsById = Object.fromEntries( ( templateParts || [] ).map( p => [ p.id, p ] ) );
```